File: leetcode/c++/9.Palindrome.cpp
#include <iostream>

using namespace std;

class Solution {
    public:
     static bool isPalindrome(int x) {
        int n=x;
        if(x<0){
          return false;
        }
        int rev=0;
        while(n>0){
          rev= rev*10 + n%10;
          n/=10;
        }
        if(x==rev){
          return true;
        }else{
          return false;
        }
    }
};

int main() {
  cout << boolalpha;
  cout << Solution::isPalindrome(-121) <<endl;
  return 0;
}

File: leetcode/js/9.Palindrome.js
class Solution{
    static isPalindrome(x){
      var n=x
      if(n<0){
        return false
      }
      var rev=0
      while(parseInt(n)>0){
        rev=rev*10+parseInt(n)%10
        n/=10
      }
      if(rev==x){
        return true
      }else{
        return false
      }
    } 
 }
 
 console.log(Solution.isPalindrome(121))